src/rust/ide/src/model/suggestion_database.rs

```diff
@@ -230,6 +266,11 @@ impl SuggestionDatabase {
         self.entries.borrow().get(&id).cloned()
     }
 
+    /// Get suggestion entry by id. Returns `Result`.
+    pub fn lookup(&self, id:EntryId) -> Result<Rc<Entry>,NoSuchEntry> {
+        self.get(id).ok_or(NoSuchEntry(id))
+    }
+
```

Comment:
Other "lookup" methods could return Result instead of Option as well.

Reply:
I've looked into this but come to a conclusion that this isn't as good idea.
Other lookup methods do not look for the `EntryId` but rather try matching by some arbitrary criterais. As such, not being able to match any entry in such cases is much less error than in this case. 
Also, all usages would just discard such errors. And each of them would require custom error types.
Too much work with no use.